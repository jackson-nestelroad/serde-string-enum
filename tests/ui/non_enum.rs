use serde_string_enum::{DeserializeStringEnum, SerializeStringEnum};

#[derive(SerializeStringEnum, DeserializeStringEnum)]
struct Type {
    name: String,
}

fn main() {}
