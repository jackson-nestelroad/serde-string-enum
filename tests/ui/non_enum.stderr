error: input must be an enum
 --> tests/ui/non_enum.rs:3:10
  |
3 | #[derive(SerializeStringEnum, DeserializeStringEnum)]
  |          ^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `SerializeStringEnum` (in Nightly builds, run with -Z macro-backtrace for more info)

error: input must be an enum
 --> tests/ui/non_enum.rs:3:31
  |
3 | #[derive(SerializeStringEnum, DeserializeStringEnum)]
  |                               ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `DeserializeStringEnum` (in Nightly builds, run with -Z macro-backtrace for more info)
